### PR TITLE
V2.0.0

### PR DESCRIPTION
**Pull request by EmmaPepin**
Can you check and notify me if you find suspect thing or bug, thx.

### CODE REVIEW

File: src/scss/blocks/_button.scss [hunk 1]
```diff
@@ -133,160 +103,216 @@
...
+            box-shadow: 0 0 $size-2 $color, inset 0 0 $size-2 $color;
+        }
     }
 
-    &:disabled .button__icon,
-    &[disabled="disabled"] .button__icon,
-    &[aria-disabled="true"] .button__icon,
-    &.button-disabled .button__icon {
-        fill: $grey-65;
+    .button__label {
```
Comment (CrucifixArnaud) [inline src/scss/blocks/_button.scss]: The vertical alignment of button (with and without icon) is a little bit strange (if you check the doc page), maybe we should refactor the way we build this aspect of the button.

File: src/scss/utils/_variables.scss [hunk 2]
```diff
@@ -60,28 +47,21 @@ $grey-65: #a6a6a6 !default;
...
-$grey-50: #808080 !default;
+$functional-success: #6ec880 !default;
+$functional-error: #c4262e !default;
+$functional-warning: #ebca73 !default;
 
 // Form & buttons
 // ---------------------
-$inputBorderRadius: $size-20 !default;
-$buttonBorderRadius: $inputBorderRadius !default;
+$inputBorderRadius: $size-8 !default;
```
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply src/scss/utils/_variables.scss]: Done.

File: src/scss/blocks/form/_input-radio.scss [hunk 3]
```diff
@@ -14,44 +14,45 @@
     opacity: 0;
 }
 
-.input--radio__label {
+.input--radio__label{
```
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply src/scss/blocks/form/_input-radio.scss]: Bien vu ;-) Done

File: src/scss/blocks/form/_input-checkbox.scss [hunk 4]
```diff
@@ -20,38 +20,41 @@
...
     position: absolute;
     top: 50%;
     transform: translateY(-50%);
 }
 
-
 // Fake Checkbox Active
 .input--checkbox__label:after {
-    background: url("#{$public-path}images/icons/icon-checkmark.svg") no-repeat;
+    background-image: url("#{$public-path}images/icons/icon-checkmark.svg");
```
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply src/scss/blocks/form/_input-checkbox.scss]: Done

File: src/scss/blocks/form/_input-radio.scss [hunk 5]
```diff
@@ -69,21 +70,21 @@
...
 
-// .input--radio__input:checked:focus + .input--radio__label:before {
-//     border-color: $primary-dd;
-// }
-
-// Disabled State
-.input--radio__input:disabled + .input--radio__label {
-    color: fade-out(#000, .8);
-    cursor: not-allowed;
+// Error State
```
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply src/scss/blocks/form/_input-radio.scss]: Already done.

File: src/scss/utils/_sizes.scss [hunk 6]
```diff
@@ -12,7 +12,9 @@ $size-50: 3.125rem;
 $size-48: 3rem;
 $size-44: 2.75rem;
 $size-40: 2.5rem;
+$size-38: 2.375rem;
 $size-36: 2.25rem;
```
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply src/scss/utils/_sizes.scss]: Inderdaad ;-) I have already checked all sizes but I skipped $size-36 :-( Done

File: src/scss/utils/_variables.scss [hunk 7]
```diff
@@ -13,45 +13,32 @@ $line-height-body: 1.5 !default;
...
+$primary-blue-l: #81a0c5 !default;
+$primary-blue-ll: #c4d3e4 !default;
+$primary-blue-lll: #ebf0f6 !default;
+$primary-blue-llll: #f7f9fb !default;
+
+$primary-green-l: #9be5b5 !default;
+$primary-green-ll: #cff3dc !default;
+$primary-green-lll: #eefbf2 !default;
+
+// Need to be validated by Anne-Fleur
```
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply src/scss/utils/_variables.scss]: Checked with Anne-Fleur, to be removed, done

File: src/scss/blocks/form/_input-switch.scss [hunk 8]
```diff
@@ -9,7 +9,7 @@
     font-weight: $font-weight-title;
```
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply src/scss/blocks/form/_input-switch.scss]: Inderdaad ;-) Done

File: docs/css/components/forms.md [hunk 9]
```diff
@@ -18,6 +18,26 @@ A field is the grouping of a `label` and an `input`. Enclose those within a div
 </div>
 ```
 
+## Field modal theme
```
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply docs/css/components/forms.md]: Done by CrucifixArnaud.

File: src/scss/blocks/_button.scss (hunk 1 above)
Comment (EmmaPepin) [reply src/scss/blocks/_button.scss]: Done by CrucifixArnaud.

File: src/scss/blocks/form/_input-group.scss [hunk 10]
```diff
@@ -119,49 +105,40 @@
...
-    right: -2px;
-}
-
-.input-group--stepper__button--append:focus::after {
-    left: -2px;
-}
-
 // Select
 // ---------------------
 .input-group--select {
```
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply src/scss/blocks/form/_input-group.scss]: Input group stepper added and input from to removed.

File: src/scss/blocks/form/_input-group.scss [hunk 11]
```diff
@@ -119,49 +105,40 @@
 .input-group.focused .input-group__to,
 .input-group.focused .input-group__from-to-label,
 .input-group.focused .input-group--stepper__button {
-    border-color: $primary-dd;
+    border-color: $primary-blue-lll;
 }
 
 // Stepper
 .input-group--stepper__main {
```
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply src/scss/blocks/form/_input-group.scss]: Input group stepper added and input from to removed

File: src/scss/blocks/form/_form.scss [hunk 12]
```diff
@@ -132,14 +152,14 @@
     width: calc(50% - 10px);
 }
 
-// Field Error
+// State messages
 // ---------------------
-.field__error {
+%field_message {
```
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply src/scss/blocks/form/_form.scss]: Done

File: src/scss/blocks/form/_input-select.scss [hunk 13]
```diff
@@ -178,24 +178,37 @@
...
     &:not(:checked) {
-        color: $primary-dd; /* Prevent <option>s from becoming transparent due to the focusring transparent rules */
+        color: $primary-blue-dd; /* Prevent <option>s from becoming transparent due to the focusing transparent rules */
     }
 }
 
+.input--select--mobile optgroup {
+    color: $secondary-grey;
+    font-weight: $font-weight-body;
+    background: $color-background;
```
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply src/scss/blocks/form/_input-select.scss]: Adapted according to remarks.

File: src/scss/blocks/form/_input-select.scss [hunk 14]
```diff
@@ -122,10 +121,11 @@
...
-    font-weight: $font-weight-title;
-    font-family: $font-title;
-    font-size: $size-12;
+    color: $secondary-grey;
+    font-weight: $font-weight-body;
+    font-family: $font-body;
+    font-size: $size-14;
+    line-height: $size-20;
     pointer-events: none;
     background: $color-background;
```
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply src/scss/blocks/form/_input-select.scss]: Adapted according to remarks.

File: docs/css/components/forms.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (EmmaPepin) [reply docs/css/components/forms.md]: Right suggestion, done.

**Pull request merged**